rules/why-upgrade-to-latest-dotnet/rule.md

```diff
@@ -0,0 +1,61 @@
+---
+type: rule
+archivedreason: 
+title: Do you know why to upgrade from .NET Framework to the latest .NET?
+guid: 75e296c8-1c21-4a4e-a2fb-4df32fd6140e
+uri: why-upgrade-to-latest-dotnet
+created: 2023-10-06T23:18:38.515Z
+authors:
+  - title: Yazhi Chen
+    url: https://www.ssw.com.au/people/yazhi-chen
+
+---
+In the world of software development, staying up-to-date with technology is not just a trend; it's a necessity. As the digital landscape evolves, so do the tools and frameworks that developers rely on to build applications. One significant transition that has been taking place in recent years is the migration from the legacy .NET Framework to the latest .NET offerings. Read more about [.NET version lifecycle](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#lifecycle). But why is this shift so crucial, and what benefits does it bring to the table?
+<!--endintro-->
+
+## The Evolution of .NET
+
+To understand the significance of upgrading from .NET Framework to the latest .NET, let's first take a brief look at the evolution of the .NET platform.
+
+### .NET Framework: The Legacy
+
+The .NET Framework has been the backbone of Windows application development for nearly two decades. It has provided a robust and versatile environment for building Windows desktop applications, web applications, and services. However, as technology progressed, the limitations of .NET Framework became more apparent.
+
+#### The Pain Point:
+
+* Compatibility - .NET Framework applications run on Windows, limiting cross-platform deployment.
+* Performance - Each new version of .NET brings improved performance and resource utilization, allowing the same .NET code to perform better on the same hardware.
+* Security - Limited security support for older .NET Framework versions(4,4.5 and 4.5.1), and the support for NET Framework versions(4.5.2 and later) [follows the lifecycle policy of the underlying Windows OS on which it is installed](https://learn.microsoft.com/en-us/lifecycle/faq/dotnet-framework#what-is-the-microsoft-lifecycle-policy-for--net-framework-).
+* Modern development features - Newer versions of .NET give us more opportunities to design products and services using many newer capabilities that are not backported to .NET Framework. This allows us to build, test and deploy faster, and more securely than ever before, keeping us competitive in today's tech landscape.
+
+### Enter the Modern .NET
+
+Recognizing the need for change, Microsoft introduced a new direction for .NET with the release of .NET Core, which later evolved into .NET 5 and beyond. This modern .NET is designed to address the shortcomings of .NET Framework and meet the demands of contemporary software development.
+
+## Key Reasons to upgrade from .NET Framework to the Latest .NET
+
+### Cross-Platform Compatibility
```

Comment:
@YazhiChen these should be h4's, h3 doesn't style right on the pages

Reply:
Hi @GordonBeeming, this is fixed in the PR #6959